saito-wasm/src/wasm_transaction.rs

```diff
@@ -32,10 +34,16 @@ impl WasmTransaction {
 
     #[wasm_bindgen(getter = routing_path)]
     pub fn get_routing_path(&self) -> Array {
-        let array = Array::new();
-        for hop in &self.tx.path {
-            let wasm_hop = WasmHop::from_hop(hop.clone());
-            array.push(&JsValue::from(wasm_hop));
+        let mut paths: Vec<WasmHop> = self
+            .tx
+            .path
+            .iter()
+            .map(|path| WasmHop::from_hop(path.clone()))
+            .collect();
+
+        let array = js_sys::Array::new_with_length(paths.len() as u32);
```

Comment:
what changed here?

Reply:
I did some refactoring

Reply:
but it now has 2 loops instead of one. right ?